js/src/common/components/Modal.js

```diff
@@ -10,10 +10,28 @@ import Button from './Button';
  */
 export default class Modal extends Component {
   /**
-   * Determine whether or not the modal should be dismissible via an 'x' button.
+   * Determine whether or not the modal should be dismissible.
+   *
+   * This overrides `doesBackdropClickDismiss` and `doesEscKeyDismiss`.
    */
   static isDismissible = true;
 
+  /**
+   * Determine whether a click on the modal backdrop should dismiss the modal.
+   *
+   * Clicking and dragging from certain elements onto the backdrop can sometimes
+   * inadvertantly trigger this.
+   *
+   * It's recommended to disable this for modals which contain forms that may
+   * take a user a longer amount of time to complete to prevent frustration.
+   */
+  static doesBackdropClickDismiss = false;
```

Comment:
How do we feel about this wording vs `dismissOnBackdropClick`?

Reply:
That name sounds way better in my opinion. I think we should go for it.